### PR TITLE
Added Editor (Quill Editor) Field

### PR DESCRIPTION
**Pull request by markvesterskov**
I have added a new field type, Editor, which is a simple implementation of the QuillJS editor.

For now the field is not customizable, but it wouldn't be impossible to add options to customize Quill if needed.

Since i have no experience writing tests, there are none included sorry.

I'd love feedback on this, and any ideas for changes and improvements.

### CODE REVIEW

File: src/Http/Controllers/SettingsToolController.php [hunk 1]
```diff
@@ -26,7 +26,9 @@ public function read()
             ->flatMap(function ($panel) use ($settings) {
                 return [$panel => $settings->where('panel', $panel)->pluck('key')->all()];
             })
-            ->merge(['_default' => $settings->where('panel', null)->pluck('key')->all()])
+            ->when($settings->where('panel', null)->isNotEmpty(),function($collection) use ($settings){
+                return $collection->merge(['_default' => $settings->where('panel', null)->pluck('key')->all()]);
+            })
```
Comment (bakerkretzmar) [inline src/Http/Controllers/SettingsToolController.php:31]: 🤦🏻‍♂️ thanks for catching this!

**Comment by bakerkretzmar** [comment]
Thanks for the PR! Is there a specific situation you need this for? I'm not a fan of WYSIWYG editors, Quill's really nice but my gut reaction is that this is overkill and you can use a `textarea` or `code` HTML field.

**Comment by markvesterskov** [comment]
Thanks for the awesome tool!

I am currently deploying a site for a new client where they needed a WYSIWYG editor for their settings. They store stuff like their footer information, and even a couple of content pages using your settings tool, and I cannot ask them to stick to plain text, and definitely not to use the code editor - regular non-programmer people, you know.

WYSIWYG editors arent exactly awesome, but sometimes there is no way around them, and it adds more options to your already really useful tool. Quill was the least cumbersome and bloated editor i could find, and only the bare minimums are imported. I've tried keeping the footprint down, but of course it adds a bit of overhead.

**Comment by ibrahem-kamal** [comment]
@bakerkretzmar is there any chance u gonna merge this request or implement this feature ? is really helpful for pages like privacy policy and terms and conditions

**Pull request closed without merging**